### ISSUE TITLE
Unique index not working with Foreign Key Reference

### ISSUE THREAD

**Issue by striker4150** [yajra/laravel-oci8#737]
### Unique index not working with Foreign Key Reference

The SQL used to create a unique index via function-based indexing prevents the index from being used as a foreign key reference.

This means that when a unique index is created using `$table->unique('unique_key')` or `$table->int('unique_key')->unique()`, `$table->foreign('key')->references('unique_key')->on('table_name')` does not work, and raises the error `ORA-02270: no matching unique or primary key for this column-list`.

### Code snippet of problem

```php
$this->schema->create('telescope_entries', function (Blueprint $table) {
    $table->bigIncrements('sequence');
    $table->uuid('uuid');
    $table->uuid('batch_id');
    $table->string('family_hash')->nullable();
    $table->boolean('should_display_on_index')->default(true);
    $table->string('type', 20);
    $table->longText('content');
    $table->dateTime('created_at')->nullable();

    $table->unique('uuid');
    $table->index('batch_id');
    $table->index('family_hash');
    $table->index('created_at');
    $table->index(['type', 'should_display_on_index']);
});

$this->schema->create('telescope_entries_tags', function (Blueprint $table) {
    $table->uuid('entry_uuid');
    $table->string('tag');

    $table->index(['entry_uuid', 'tag']);
    $table->index('tag');

    $table->foreign('entry_uuid')
        ->references('uuid')  //  ORA-02270: no matching unique or primary key for this column-list
        ->on('telescope_entries')
        ->onDelete('cascade');
});
```

### System details

- Windows 10
- PHP 7.4.30
- Laravel Version 8.x
- Laravel-OCI8 Version 8.6.2

**Comment by striker4150** [comment]
The issue #709 also brought up this problem, but was marked as stale.

**Comment by striker4150** [comment]
Parent Table:
```
create table telescope_entries ( "SEQUENCE" number(19,0) not null, uuid char(36) not null, batch_id char(36) not null, family_hash varchar2(255) null, should_display_on_index char(1) default '1' not null, type varchar2(20) not null, content clob not null, created_at date null, constraint telescope_entries_sequence_pk primary key ( "SEQUENCE" ) )
```

Child Table (error):
```
Error Message : ORA-02270: no matching unique or primary key for this column-list
Position      : 192
Statement     : create table telescope_entries_tags ( entry_uuid char(36) not null, tag varchar2(255) not null, constraint telesco_entri_ta_ent_uu_fk foreign key ( entry_uuid ) references telescope_entries ( uuid ) on delete cascade )
```

**Comment by striker4150** [comment]
Bump

**Comment by striker4150** [comment]
@yajra Sorry to disturb you, but could I please get an update on this? If there is something that I am missing, or this issue cannot be fixed, please let me know. This is preventing me from moving my Laravel Telescope migrations into my Oracle DB.

**Comment by yajra** [comment]
Sorry, overlooked these issues. Should we revert https://github.com/yajra/laravel-oci8/pull/599? Can you please give it a try?

**Comment by striker4150** [comment]
I just confirmed that the Laravel Telescope default migration works on [v8.1.0](https://github.com/yajra/laravel-oci8/releases/tag/v8.1.0) and not on [v8.1.1](https://github.com/yajra/laravel-oci8/releases/tag/v8.1.1).

**Comment by yajra** [comment]
Fixed on v8.6.3, please check and confirm. Thanks!

**Comment by striker4150** [comment]
Yup, the revert fixed it. Thanks!
